### PR TITLE
Identity Creation: Milestone1

### LINKED ISSUE #1: Identity Creation Extrinsic
A function will need to be defined in the fennel-identity pallet to create an Identity. This will function as a single transaction that other identity transactions point to as their root.

### PR DESCRIPTION
**Pull request by isaacadams**
closes #1

### CODE REVIEW

**Review by Romulus10 — CHANGES_REQUESTED** [review]
I'm going to open a new issue RE: the Trust pallet. You made a couple really good storage-safety decisions here that I haven't gotten to implementing yet.

We're looking good other than the few adjustments I mention in these comments.

File: pallets/identity/src/lib.rs [hunk 1]
```diff
@@ -58,6 +71,19 @@ pub mod pallet {
...
+
+            let total_ids: u32 = <IdentityNumber<T>>::get();
+            let new_id: u32 = total_ids.checked_add(1).ok_or(Error::<T>::StorageOverflow)?;
+
+            <IdentityList<T>>::try_mutate(&who, |ids| -> DispatchResult {
+                ids.insert(new_id);
+                Ok(())
+            })?;
+
+            <IdentityNumber<T>>::put(new_id);
```
[the remark this answers is not in the document]
Comment (isaacadams) [reply pallets/identity/src/lib.rs:84]: won't with the `?` on line 82 cause it to return?

File: pallets/identity/src/lib.rs [hunk 2]
```diff
@@ -58,6 +71,19 @@ pub mod pallet {
         #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
         /// Create a new identity owned by origin.
         pub fn create_identity(origin: OriginFor<T>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            let total_ids: u32 = <IdentityNumber<T>>::get();
+            let new_id: u32 = total_ids.checked_add(1).ok_or(Error::<T>::StorageOverflow)?;
+
+            <IdentityList<T>>::try_mutate(&who, |ids| -> DispatchResult {
```
[the remark this answers is not in the document]
Comment (isaacadams) [reply pallets/identity/src/lib.rs:79]: if StorageOverflow occurs, it will return without affecting storage + if try mutate fails, the `?` on line 82 should cause a return without changing state

File: pallets/identity/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Romulus10) [reply pallets/identity/src/lib.rs:84]: Missed the ?, my mistake.

**Review by Romulus10 — COMMENTED** [review]
I think there are a few more places we can test this - starting with trying to overflow that StorageValue and getting it to fail gracefully. I'll do the same over in the Trust pallet.

File: pallets/identity/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Romulus10) [reply pallets/identity/src/lib.rs:79]: I think I may have just straight-up forgotten how `?` works for a second there. I'll chalk it up to not using Rust much for a while. 😂 Gave us a really good chance to talk out some improvements to both pallets, though.

**Review by Romulus10 — APPROVED** [review]
